### ISSUE TITLE
pre-release packages no longer gets verified

### ISSUE THREAD

**Issue by mkevenaar** [chocolatey/package-verifier#89]
After implementing #86 / #87 pre-release packages no longer gets verified when failing the validation tests.

https://chocolatey.org/packages/firefox-beta/80.0.3-beta

Would it be possible to exempt pre-release packages from this rule?

**Comment by pauby** [comment]
Should it not be the case every package should pass the Validator before being Verified? In this case the Validation failed because the package has the word `Beta` in it (which was added [here](https://github.com/chocolatey/package-validator/issues/92)).

I think this is more of a Validator issue than a Verifier one.

/cc @gep13

**Comment by mkevenaar** [comment]
@pauby would you say in case of a pre-release package, if the validator failed it should automatically mark the validation step as exempted?

I am not sure I would call it a "Validator issue" as this is a direct result of implementing #86 

In this case, it's actually called "Firefox Beta":

![image](https://user-images.githubusercontent.com/834643/89430332-eea5aa00-d73e-11ea-8693-f1d873a24381.png)

https://www.mozilla.org/en-US/firefox/all/#product-desktop-release

**Comment by pauby** [comment]
@pauby would you say in case of a pre-release package, if the validator failed it should automatically mark the validation step as exempted?

No. A package should always pass the Validator steps (exceptions excluded). A pre-release package should still pass the Validator checks _in my opinion_.

> I am not sure I would call it a "Validator issue" as this is a direct result of implementing #86

So previously this would fail the validator and still be approved as 'Exempted'?

> In this case, it's actually called "Firefox Beta":

The package has `-beta` in the name which is what I was referring to. One of the Validator rules checks if a package name includes that.

**Comment by AdmiringWorm** [comment]
> So previously this would fail the validator and still be approved as 'Exempted'?

yes, it would.
Even if the validator (or verifier) failed, all pre-release packages would still be made exempted.

**Comment by pauby** [comment]
Then I agree with what's in the original comment, and take back what I said.

I do however still believe that these packages should fail the Validator. If they did we wouldn't have a `firefox-beta` package for example (which is against the Validator rules) but because they go through anyway, the rules that are meant to stop a package having `-beta` in them go through - that's a loophole.

I know I'm picking on you @mkevenaar and your package, and also on that one rule, but it equally applies to all the other packages and all the other rules. I know you can take it though 😄 I may raise an issue about this tomorrow as it's separate issue entirely.

**Comment by AdmiringWorm** [comment]
> I do however still believe that these packages should fail the Validator.

I completely agree in general, however this *loophole* have been actively used for several years now, and I would be a bit hesitant to say that it should be outright enabled.

I would instead suggest to prevent any new packages using a pre-release id to result in a push error instead (not new version, only new packages).

**Comment by ferventcoder** [comment]
I'd consider this a higher priority for sure to resolve.

**Comment by gep13** [comment]
@mkevenaar @pauby @AdmiringWorm @ferventcoder prior to the change that was made in this [PR](https://github.com/chocolatey/package-verifier/pull/87), a pre-release package, regardless of the outcome of the package validator and package verifier, would be moved to exempted automatically.  With the change in place, this doesn't happen anymore, since a package will not be moved into the package verifier queue unless it has passed the package validator, or been exempted from it.

Going forward, we will need to change the query that is completed by both the package verifier, and package scanner, to include packages that have failed the previous service check _and_ is a pre-release package.

With this change in place, it will mean that package maintainers still get to _see_ what services were passed/failed, so that when they switch to a non pre-release version, they need to know what has been addressed.
